tests/client/test_proxies.py

```diff
@@ -244,3 +247,21 @@ def test_proxies_environ(monkeypatch, client_class, url, env, expected):
         assert transport == client._transport
     else:
         assert transport.proxy_origin == url_to_origin(expected)
+
+
+@pytest.mark.parametrize(
+    ["proxies", "expected_scheme"],
+    [
+        ({"http": "http://127.0.0.1"}, ["http://"]),
+        ({"https": "http://127.0.0.1"}, ["https://"]),
+        ({"all": "http://127.0.0.1"}, ["all://"]),
+    ],
+)
+def test_for_deprecated_proxy_params(proxies, expected_scheme):
+    with pytest.deprecated_call() as block:
+        httpx.AsyncClient(proxies=proxies)
+
+    warning_message = str(block.pop(DeprecationWarning))
+
+    for scheme in expected_scheme:
+        assert scheme in warning_message
```

Comment:
Could this be simplified to this by converting `expected_scheme` to a plain string?

```suggestion
    assert expected_scheme in warning_message
```

Reply:
Yeah I think that was walking through the *characters* in the string. :)

Reply:
No, it was correct (I expect CI to fail now! ;-)), but `expected_scheme` was unecessarily a 1-item list. Sent a patch…